filetests/isa/x86/pshuf.clif

```diff
@@ -0,0 +1,11 @@
+test binemit
+; set opt_level=best
+target x86_64
+
+function %test_pshuf() {
+ebb0:
+[-, %rax]   v0 = iconst.i32 42
+[-, %xmm0]  v1 = bitcast.i32x4 v0    ; bin: 66 40 0f 6e c0
+[-, %xmm0]  v2 = x86_pshuf v1, 0     ; bin: 66 0f 70 c0 00
+            return
```

Comment:
So this should now be semi-correct; I verified on my machine that this is what I would expect a MOVD + PSHUFD to look like. I say semi-correct because GCC on my machine expands the REX byte (the `40` in the bitcast) to a `48` so as to use 64 bits instead of 32. Because this instruction operates on 32 bit values it seems like we could drop the REX byte entirely here; unfortunately `enc_both` adds the REX encoding and this must take precedence (?).

Reply:
What I mean is: should the encoder be smarter and pick space-saving encodings when possible? I see a comment at x86/encodings.rs:164: `REX-less encoding must come after REX encoding so we don't use it by default. Otherwise reg-alloc would never use r8 and up.` Is this something that can be fixed or that's just the way it is? Or does anyone even care since it is just one extra byte?

Reply:
More info on the REX byte in section of 2.2.1 of Intel Software Developer's Manual (volume 2).

Reply:
As said on IRC: by un-commenting line 2, you should get the smallest encoding for this instruction.